README.md

```diff
@@ -249,6 +249,26 @@ $executor = new RetryExecutor(
 );
 ```
 
+Note that this executor is entirely async and as such allows you to execute
+any number of queries concurrently. You should probably limit the number of
+concurrent queries in your application or you're very likely going to face
+rate limitations and bans on the resolver end. For many common applications,
+you may want to avoid sending the same query multiple times when the first
+one is still pending, so you will likely want to use this in combination with
+a `CoopExecutor` like this:
+
+```php
+$executor = new CoopExecutor(
```

Comment:
Maybe `RequestSharingExecutor` / `QuerySharingExecutor` is a better name?

Reply:
Maybe. I don't have a strong preference for either name, so I'll leave this up to somebody else to decide if somebody has a strong preference and/or some good arguments :+1: